src/plugins/data/public/search/search_source/search_source.ts

```diff
@@ -184,6 +185,34 @@ export class SearchSource {
     return this.parent;
   }
 
+  /**
+   * Run a search using the search service
+   * @return {Promise<SearchResponse<unknown>>}
+   */
+  private runSearch(
```

Comment:
Nit: I'm not a huge fan of the name `runSearch`. Since we want to eventually add a method that actually returns the observable rather than the promise, what if we renamed this to `fetch$` and returned the observable instead? Then, in the `fetch` method we could essentially just return `fetch$().toPromise()`. Thoughts?

Reply:
Done!